python-packages/order_utils/src/zero_ex/order_utils/__init__.py

```diff
@@ -107,47 +82,154 @@ class SignatureType(Enum):
+     'makerAddress': '0x0000000000000000000000000000000000000000',
+     'makerAssetAmount': 1,
+     'makerAssetData': '0x0000000000000000000000000000000000000000',
+     'makerFee': 0,
+     'salt': 1,
+     'senderAddress': '0x0000000000000000000000000000000000000000',
+     'takerAddress': '0x0000000000000000000000000000000000000000',
+     'takerAssetAmount': 1,
+     'takerAssetData': '0x0000000000000000000000000000000000000000',
+     'takerFee': 0}
+    """
+    jsdict = cast(Dict, copy(order))
+
+    # encode bytes fields
+    jsdict["makerAssetData"] = "0x" + order["makerAssetData"].hex()
+    jsdict["takerAssetData"] = "0x" + order["takerAssetData"].hex()
+
+    jsdict["exchangeAddress"] = exchange_address
+
+    assert_valid(jsdict, "/orderSchema")
+
+    return jsdict
+
+
+def jsdict_order_to_struct(jsdict: dict) -> Order:
+    r"""Convert a JSON-schema-compatible dict order to a Web3-compatible struct.
+
+    More specifically, do explicit encoding of the `bytes` fields.
+
+    >>> import pprint
+    >>> pprint.pprint(jsdict_order_to_struct(
+    ...     {
+    ...         'makerAddress': "0x0000000000000000000000000000000000000000",
+    ...         'takerAddress': "0x0000000000000000000000000000000000000000",
+    ...         'feeRecipientAddress': "0x0000000000000000000000000000000000000000",
+    ...         'senderAddress': "0x0000000000000000000000000000000000000000",
+    ...         'makerAssetAmount': 1000000000000000000,
+    ...         'takerAssetAmount': 1000000000000000000,
+    ...         'makerFee': 0,
+    ...         'takerFee': 0,
+    ...         'expirationTimeSeconds': 12345,
+    ...         'salt': 12345,
+    ...         'makerAssetData': "0x0000000000000000000000000000000000000000",
+    ...         'takerAssetData': "0x0000000000000000000000000000000000000000",
+    ...         'exchangeAddress': "0x0000000000000000000000000000000000000000",
+    ...     },
+    ... ))
+    {'expirationTimeSeconds': 12345,
+     'feeRecipientAddress': '0x0000000000000000000000000000000000000000',
+     'makerAddress': '0x0000000000000000000000000000000000000000',
+     'makerAssetAmount': 1000000000000000000,
+     'makerAssetData': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
+                       b'\x00\x00\x00\x00\x00\x00\x00\x00',
+     'makerFee': 0,
+     'salt': 12345,
+     'senderAddress': '0x0000000000000000000000000000000000000000',
+     'takerAddress': '0x0000000000000000000000000000000000000000',
+     'takerAssetAmount': 1000000000000000000,
+     'takerAssetData': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
+                       b'\x00\x00\x00\x00\x00\x00\x00\x00',
+     'takerFee': 0}
+    """  # noqa: E501 (line too long)
+    assert_valid(jsdict, "/orderSchema")
+
+    order = cast(Order, copy(jsdict))
+
+    def strip_0x(hex_string: str) -> str:
+        if hex_string.startswith("0x"):
+            return hex_string[2:]
+        return hex_string
```

Comment:
Why have this be a nested method? I think we will need this in other places. Can we move it to `utils`?

Reply:
Did it because I default to as-local-as-possible, unless a specific, known re-use is intended.  But I see your point, and, checking further, I see that there's an `eth_utils.remove_0x_prefix()` method that I can use instead of re-inventing the wheel, so I'll do that.